src/components/modals/DeleteExternalSourceModal.svelte

```diff
@@ -0,0 +1,113 @@
+<svelte:options immutable={true} />
+
+<script lang="ts">
+  import { base } from '$app/paths';
+  import { createEventDispatcher } from 'svelte';
+  import { plans } from '../../stores/plans';
+  import type { ExternalSourcePkey, ExternalSourceSlim } from '../../types/external-source';
+  import Collapse from '../Collapse.svelte';
+  import Modal from './Modal.svelte';
+  import ModalContent from './ModalContent.svelte';
+  import ModalFooter from './ModalFooter.svelte';
+  import ModalHeader from './ModalHeader.svelte';
+
+  export let height: number = 200;
+  export let linked: { pkey: ExternalSourcePkey; plan_ids: number[] }[] = [];
+  export let width: number = 400;
+  export let sources: ExternalSourceSlim[] = [];
+  export let unassociatedSources: ExternalSourceSlim[] = [];
+
+  const dispatch = createEventDispatcher<{
+    close: void;
+    confirm: void;
+    refresh: void;
+  }>();
+
+  function onKeydown(event: KeyboardEvent) {
+    const { key } = event;
+    if (key === 'Enter') {
+      event.preventDefault();
+      dispatch('confirm');
+    }
+  }
+</script>
+
+<svelte:window on:keydown={onKeydown} />
+
+<Modal {height} {width}>
+  <ModalHeader on:close>
+    {#if linked.length > 0}
+      External Source Cannot Be Deleted
+    {:else}
+      Delete External Source
+    {/if}
+  </ModalHeader>
+  <ModalContent>
+    {#if linked.length > 0}
+      <span class="modal-content">
+        The following External Sources are part of Derivation Groups that are still associated with the following plans:
+
+        {#each linked as link}
```

Comment:
Can you make this modal larger and also add overflow auto behavior so that this doesn't happen?
<img width="489" alt="image" src="https://github.com/user-attachments/assets/130fbf8b-e9fc-4858-b26e-f64a676c50c9">